### PR TITLE
Move MacCatalyst arm64 leg definitions from runtime-staging to runtime-extra-platforms pipeline

### PR DESCRIPTION
**Pull request by MaximLipnin**
It's a follow-up on https://github.com/dotnet/runtime/pull/64452. 

The change is to move MacCatalyst Arm64 legs (basic and sandbox-enabled) from `runtime-staging` to `runtime-extra-platforms` pipeline. Those legs will be available for manual per-PR run as we have enough devices for that.

cc @premun

### CODE REVIEW

File: eng/pipelines/runtime-extra-platforms-other.yml [hunk 1]
```diff
@@ -183,7 +183,7 @@ jobs:
     runtimeFlavor: mono
     platforms:
       - MacCatalyst_x64
-      # - MacCatalyst_arm64 # re-enable it here and remove from runtime-staging after it's stabilized on OSX 11.00 queue
+      - MacCatalyst_arm64
```
Comment (mdh1418) [inline eng/pipelines/runtime-extra-platforms-other.yml:186]: Do we guard this with `isRollingBuild` as well until we get more devices?

Comment (MaximLipnin) [reply eng/pipelines/runtime-extra-platforms-other.yml:186]: Good point, thanks! @steveisok @akoeplinger @premun Do you happen to know whether there is still lack of devices for that?

Comment (MaximLipnin) [reply eng/pipelines/runtime-extra-platforms-other.yml:186]: Discussed with @akoeplinger - there is enough device capacity at the moment, so we can try to make both arm64 legs available for manual per-PR run.

**Review by akoeplinger — APPROVED** [review]

**Review by steveisok — APPROVED** [review]

**Comment by MaximLipnin** [comment]
The failures are unrelated and known, Android arm64 one seems to be infra-related, I asked core-eng folks to look at the machines.

**Pull request merged**